climada/hazard/io.py

```diff
@@ -889,10 +893,135 @@ def vshape(array):
+        (and convert to string otherwise), although other checks and casting could be included here in the future.
+
+        Parameters
+        ----------
+
+        attrs : dict
+            Attributes for a new Hazard object
+
+        Returns
+        -------
+
+        attrs : dict
+            Attributes checked for type validity and casted otherwise (only event_name at the moment).
+
+        Warns
+        -----
+
+        UserWarning
+            Warns the user if any value casting happens.
+        """
+
+        def _check_and_cast_container(
+            attr_value: Any, expected_container: Collection
+        ) -> Any:
+            """Check if the attribute is of the expected container type and cast if necessary.
+
+            Parameters
+            ----------
+            attr_value : any
+                The current value of the attribute.
+
+            expected_container : type
+                The expected type of the container (e.g., list, np.ndarray).
+
+            Returns
+            -------
+            attr_value : any
+                The value cast to the expected container type, if needed.
+            """
+            if not isinstance(attr_value, expected_container):
+                warnings.warn(
+                    f"Value should be of type {expected_container}. Casting it.",
+                    UserWarning,
+                )
+                # Attempt to cast to the expected container type
+                if expected_container is list:
+                    return list(attr_value)
+                elif expected_container is np.ndarray:
+                    return np.array(attr_value)
+                else:
+                    raise TypeError(f"Unsupported container type: {expected_container}")
+            return attr_value
+
+        def _check_and_cast_elements(
+            attr_value: Any, expected_dtype: Union[Any, None]
+        ) -> Any:
+            """Check if the elements of the container are of the expected dtype and cast if necessary,
+            while preserving the original container type.
+
+            Parameters
+            ----------
+            attr_value : any
+                The current value of the attribute (a container).
+
+            expected_dtype : type or None
+                The expected type of the elements within the container. If None, no casting is done.
+
+            Returns
+            -------
+            attr_value : any
+                The value with elements cast to the expected type, preserving the original container type.
+            """
+            if expected_dtype is None:
+                # No dtype enforcement required
+                return attr_value
+
+            container_type = type(attr_value)  # Preserve the original container type
+
+            # Perform type checking and casting of elements
+            if isinstance(attr_value, (list, np.ndarray)):
+                if not all(isinstance(val, expected_dtype) for val in attr_value):
```

Comment:
👍 good choice. even though in case of `attr_value` being a `nd.array`, `np.issubtye(attr_value, expected_type)` would be infinitely faster but make the code more complex with nothing to be gained but a couple of seconds if the array is huge.